pkg/kubernetes/podtemplatespec/v1alpha1/podtemplatespec.go

```diff
@@ -0,0 +1,278 @@
+			b.errs,
+			errors.New(
+				"failed to build podtemplatespec object: missing annotations",
+			),
+		)
+		return b
+	}
+
+	// copy of original map
+	newannotations := map[string]string{}
+	for key, value := range annotations {
+		newannotations[key] = value
+	}
+
+	// override
+	b.podtemplatespec.Object.Annotations = newannotations
+	return b
+}
+
+// WithLabelsNew sets the labels field of podtemplatespec
+func (b *Builder) WithLabelsNew(labels map[string]string) *Builder {
+	if len(labels) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New(
+				"failed to build podtemplatespec object: missing labels",
+			),
+		)
+		return b
+	}
+
+	// copy of original map
+	newlbls := map[string]string{}
+	for key, value := range labels {
+		newlbls[key] = value
+	}
+
+	// override
+	b.podtemplatespec.Object.Labels = newlbls
+	return b
+}
+
+// WithNodeSelectorNew sets the nodeselector field of podtemplatespec
+func (b *Builder) WithNodeSelectorNew(nodeselectors map[string]string) *Builder {
+	if len(nodeselectors) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New(
+				"failed to build podtemplatespec object: missing nodeselectors",
+			),
+		)
+		return b
+	}
+
+	// copy of original map
+	newnodeselectors := map[string]string{}
+	for key, value := range nodeselectors {
+		newnodeselectors[key] = value
+	}
+
+	// override
+	b.podtemplatespec.Object.Spec.NodeSelector = newnodeselectors
+	return b
+}
+
+// WithServiceAccountName sets the ServiceAccountnNme field of podtemplatespec
+func (b *Builder) WithServiceAccountName(serviceAccountnNme string) *Builder {
+	if len(serviceAccountnNme) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New(
+				"failed to build podtemplatespec object: missing serviceaccountname",
+			),
+		)
+		return b
+	}
+
+	b.podtemplatespec.Object.Spec.ServiceAccountName = serviceAccountnNme
+	return b
+}
+
+// WithAffinity sets the affinity field of podtemplatespec
+func (b *Builder) WithAffinity(affinity *corev1.Affinity) *Builder {
+	if affinity == nil {
+		b.errs = append(
+			b.errs,
+			errors.New(
+				"failed to build podtemplatespec object: missing affinity",
+			),
+		)
+		return b
+	}
+
+	// copy of original pointer
+	newaffinitylist := *affinity
+
+	b.podtemplatespec.Object.Spec.Affinity = &newaffinitylist
+	return b
+}
+
+// WithTolerationsNew sets the tolerations field of podtemplatespec
+func (b *Builder) WithTolerationsNew(tolerations ...corev1.Toleration) *Builder {
+	if len(tolerations) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New(
+				"failed to build podtemplatespec object: missing tolerations",
+			),
+		)
+		return b
+	}
+
+	// copy of original slice
+	newtolerations := []corev1.Toleration{}
+	newtolerations = append(newtolerations, tolerations...)
+
+	b.podtemplatespec.Object.Spec.Tolerations = newtolerations
+
+	return b
+}
+
+// WithContainerBuilders builds the list of containerbuilder
+// provided and sets the containers field of the podtemplatespec
+func (b *Builder) WithContainerBuilders(
```

Comment:
Does this merge or adds. Does this result in duplicates. Please do a UT.

Reply:
Added two functions `WithContainerBuilders` and `WithContainerBuildersNew` with UT